### PR TITLE
doc: Update docs to be inline with templates

### PR DESCRIPTION
**Pull request by cprivitere**
This is a first attempt to get our docs in line with the [Kubernetes Templates](https://github.com/kubernetes/kubernetes-template-project/blob/main/README.md) as part of our effort to get the project migrated to kubernetes-sigs.

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+This is normally performed by our CI system. However, there are important steps to take first.
+
+## How to Cut a Release
+
+In order to cut a release, you must:
+
+1. Commit the changes.
+1. Push out your branch, open a PR and merge the changes
+1. Wait for the Continuous Integration github action to finish running
+1. Tag the release with `git tag -a vX.Y.z -m "Message"
```
Comment (displague) [inline RELEASE.md:16]:
```suggestion
1. Tag the release with `git tag -a vX.Y.z -m "Message"`
```

File: RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+1. Push out the tag
+
+## How A Release Happens
+
+* GitHub Actions detects a new tag has been pushed
+* CI builds docker images for each supported architecture as well as a multi-arch manifest, and tags it with the semver tag of the release, e.g. `v0.4.0`
+* CI creates the release in `out/release`, the equivalent of `make release`
```
Comment (displague) [inline RELEASE.md:23]: would we be changing our Makefile to abide by this? (no `release` task defined today)

**Review by displague — DISMISSED** [review]
lgtm - nits aside

File: LICENSE [hunk 3]
```diff
@@ -1,4 +1,192 @@
-   Copyright 2018-2021 Equinix, Inc.
+                                 Apache License
```
Comment (ctreatma) [inline LICENSE:1]: Do we need anyone to sign off on this license change?

Comment (ctreatma) [reply LICENSE:1]: Nevermind, it's the same license, your PR just adds the rest of the license text.

Comment (displague) [reply LICENSE:1]: Moving from Apache to Apache. The difference is the copyright range and name, which in this PR is updated with the LICENSE template text. We should add Kubernetes Authors too (unless there is a newer convention).

Here's an example: https://github.com/linode/linode-cloud-controller-manager/blob/master/LICENSE#L189-L192

Comment (displague) [reply LICENSE:1]: It's also pretty common, to leave this as the template text.. https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/LICENSE#L189

**Review by displague — APPROVED** [review]

**Pull request merged**